module/zfs/zfeature.c

```diff
@@ -324,6 +326,7 @@ feature_sync(spa_t *spa, zfeature_info_t *feature, uint64_t refcount,
 		VERIFY3U(*refcount_cache, ==,
 		    atomic_swap_64(refcount_cache, refcount));
 	}
+	mutex_exit(&spa->spa_feat_stats_lock);
```

Comment:
I haven't yet looked on a wider picture to propose alternatives, but it does not look nice to allow entering both with and without lock and then drop it inside.

Same time you are not protecting the activation/deactivation below.  Do you have a reason to think those are impossible?